#! /bin/sh -e
## 24_qtc_qscrollview.dpatch by Germain Garand <germain@ebooksfrance.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix the qscrollview stuff

if [ $# -lt 1 ]; then
    echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch} ${2:+-d $2}"

case "$1" in
    -patch) patch -p1 ${patch_opts} < $0;;
    -unpatch) patch -R -p1 ${patch_opts} < $0;;
    *)
        echo "`basename $0`: script expects -patch|-unpatch as argument" >&2
        exit 1;;
esac

exit 0

@DPATCH@

diff -u -3 -p -r1.65 qscrollview.cpp
--- bla/src/widgets/qscrollview.cpp	29 Apr 2004 22:31:28 -0000	1.65
+++ bla/src/widgets/qscrollview.cpp	25 Jan 2005 12:07:04 -0000
@@ -1180,7 +1180,7 @@ void  QScrollView::mouseMoveEvent( QMous
 void QScrollView::wheelEvent( QWheelEvent *e )
 {
     QWheelEvent ce( viewport()->mapFromGlobal( e->globalPos() ),
-                    e->globalPos(), e->delta(), e->state());
+                    e->globalPos(), e->delta(), e->state(), e->orientation() );
     viewportWheelEvent(&ce);
     if ( !ce.isAccepted() ) {
 	if ( e->orientation() == Horizontal && horizontalScrollBar() )
@@ -1863,7 +1863,7 @@ void QScrollView::viewportWheelEvent( QW
        the event itself.
     */
     QWheelEvent ce( viewportToContents(e->pos()),
-        e->globalPos(), e->delta(), e->state());
+        e->globalPos(), e->delta(), e->state(), e->orientation());
     contentsWheelEvent(&ce);
     if ( ce.isAccepted() )
         e->accept();
